### ISSUE TITLE
ERROR: Value does not fall within the expected range.

### ISSUE THREAD

**Issue by HenrrySalinasJala** [rgl/packer-provisioner-windows-update#25]
Hello,
I have being experiencing some issues while performing windows updates in a Win7 azure-arm type builder.
**Actual result** 
ERROR: Value does not fall within the expected range.
**Expected result**
The windows updates should be applied without errors

My template:
packer v1.3.1
windows-update v0.6.0
```
{
  "variables": {
    "client_id": "{{env `CLIENT_ID`}}",
    "client_secret": "{{env `CLIENT_SECRET`}}",
    "resource_group": "{{env `RESOURCE_GROUP`}}",
    "storage_account": "{{env `STORAGE_ACCOUNT`}}",
    "subscription_id": "{{env `SUBSCRIPTION_ID`}}",
    "vm_size": "{{env `VM_SIZE`}}",
    "saPwd": "{{env `SA_PWD`}}",
    "admin_username": "{{env `ADMIN_USERNAME`}}",
    "location": "{{env `LOCATION`}}",
    "script_folder": "{{env `SCRIPT_FOLDER`}}"
  },
  "builders": [
    {
      "type": "azure-arm",
      "client_id": "{{user `client_id`}}",
      "client_secret": "{{user `client_secret`}}",
      "resource_group_name": "{{user `resource_group`}}",
      "storage_account": "{{user `storage_account`}}",
      "subscription_id": "{{user `subscription_id`}}",
      "capture_container_name": "custom-vhd-images",
      "capture_name_prefix": "packer",
      "os_type": "Windows",
      "image_publisher": "microsoftvisualstudio",
      "image_offer": "Windows",
      "image_sku": "Win7-SP1-Ent-N-x64",
      "communicator": "winrm",
      "winrm_use_ssl": "true",
      "winrm_insecure": "true",
      "winrm_timeout": "4h",
      "winrm_username": "{{user `admin_username`}}",
      "vm_size": "{{user `vm_size`}}",
      "location": "{{user `location`}}"
    }
  ],
  "provisioners": [
    {
      "type": "powershell",
      "inline":[
        "Get-Date"
      ]
    },
    {
      "type": "windows-update"
    },
    {
      "type": "windows-restart",
      "restart_check_command": "powershell -command \"& {Write-Output 'restarted.'}\""
    },
    {
      "type": "powershell",
      "inline": [
        "if( Test-Path $Env:SystemRoot\\windows\\system32\\Sysprep\\unattend.xml ){ rm $Env:SystemRoot\\windows\\system32\\Sysprep\\unattend.xml -Force}",
        "& $env:SystemRoot\\System32\\Sysprep\\Sysprep.exe /oobe /generalize /quiet /quit",
        "while($true) { $imageState = Get-ItemProperty HKLM:\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Setup\\State | Select ImageState; if($imageState.ImageState -ne 'IMAGE_STATE_GENERALIZE_RESEAL_TO_OOBE') { Write-Output $imageState.ImageState; Start-Sleep -s 10  } else { break } }"
      ]
    }
  ]
}
```

And the output logs
```2018-10-24T01:22:08.8851399Z     azure-arm: WinRM connected.
2018-10-24T01:22:08.8852308Z 
2018-10-24T01:22:08.8888294Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:08.8889802Z ##[debug]/*************************************
2018-10-24T01:22:08.8891350Z ##[debug]    azure-arm: WinRM connected.

2018-10-24T01:22:08.8892317Z ##[debug]**************************************/
2018-10-24T01:22:08.8893312Z ##[debug]Match start index: -1
2018-10-24T01:22:08.8894477Z ##[debug]Match start index: -1
2018-10-24T01:22:08.8917546Z     azure-arm: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
2018-10-24T01:22:09.8198372Z 
2018-10-24T01:22:09.8203792Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:09.8205382Z ##[debug]/*************************************
2018-10-24T01:22:09.8207505Z ##[debug]    azure-arm: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>

2018-10-24T01:22:09.8209534Z ##[debug]**************************************/
2018-10-24T01:22:09.8210313Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8211106Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8211666Z ==> azure-arm: Connected to WinRM!
2018-10-24T01:22:09.8212007Z 
2018-10-24T01:22:09.8212678Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:09.8213521Z ##[debug]/*************************************
2018-10-24T01:22:09.8214371Z ##[debug]==> azure-arm: Connected to WinRM!

2018-10-24T01:22:09.8215924Z ##[debug]**************************************/
2018-10-24T01:22:09.8216637Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8217269Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8217756Z ==> azure-arm: Provisioning with Powershell...
2018-10-24T01:22:09.8218071Z 
2018-10-24T01:22:09.8218700Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:09.8219403Z ##[debug]/*************************************
2018-10-24T01:22:09.8220149Z ##[debug]==> azure-arm: Provisioning with Powershell...

2018-10-24T01:22:09.8220886Z ##[debug]**************************************/
2018-10-24T01:22:09.8221565Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8222190Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8222823Z ==> azure-arm: Provisioning with powershell script: C:\Users\FABuild\AppData\Local\Temp\packer-powershell-provisioner081654639
2018-10-24T01:22:09.8223254Z 
2018-10-24T01:22:09.8223888Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:09.8224599Z ##[debug]/*************************************
2018-10-24T01:22:09.8225475Z ##[debug]==> azure-arm: Provisioning with powershell script: C:\Users\FABuild\AppData\Local\Temp\packer-powershell-provisioner081654639

2018-10-24T01:22:09.8227702Z ##[debug]**************************************/
2018-10-24T01:22:09.8228359Z ##[debug]Match start index: -1
2018-10-24T01:22:09.8229211Z ##[debug]Match start index: -1
2018-10-24T01:22:19.5118218Z     azure-arm:
2018-10-24T01:22:19.5119470Z 
2018-10-24T01:22:19.5122541Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:19.5125080Z ##[debug]/*************************************
2018-10-24T01:22:19.5127713Z ##[debug]    azure-arm:

2018-10-24T01:22:19.5131416Z ##[debug]**************************************/
2018-10-24T01:22:19.5133020Z ##[debug]Match start index: -1
2018-10-24T01:22:19.5134294Z ##[debug]Match start index: -1
2018-10-24T01:22:19.5300038Z     azure-arm: Wednesday, October 24, 2018 1:22:19 AM
2018-10-24T01:22:19.5301127Z ==> azure-arm: Uploading the Windows update elevated script...
2018-10-24T01:22:19.5301678Z 
2018-10-24T01:22:19.5303954Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:19.5304757Z ##[debug]/*************************************
2018-10-24T01:22:19.5305729Z ##[debug]    azure-arm: Wednesday, October 24, 2018 1:22:19 AM
==> azure-arm: Uploading the Windows update elevated script...

2018-10-24T01:22:19.5306726Z ##[debug]**************************************/
2018-10-24T01:22:19.5307405Z ##[debug]Match start index: -1
2018-10-24T01:22:19.5308047Z ##[debug]Match start index: -1
2018-10-24T01:22:23.7977814Z ==> azure-arm: Uploading the Windows update check for reboot required elevated script...
2018-10-24T01:22:23.7979253Z 
2018-10-24T01:22:23.7980728Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:23.7982153Z ##[debug]/*************************************
2018-10-24T01:22:23.7983868Z ##[debug]==> azure-arm: Uploading the Windows update check for reboot required elevated script...

2018-10-24T01:22:23.7986426Z ##[debug]**************************************/
2018-10-24T01:22:23.7988688Z ##[debug]Match start index: -1
2018-10-24T01:22:23.7989566Z ##[debug]Match start index: -1
2018-10-24T01:22:25.8331890Z ==> azure-arm: Uploading the Windows update script...
2018-10-24T01:22:25.8332572Z 
2018-10-24T01:22:25.8333646Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:25.8334611Z ##[debug]/*************************************
2018-10-24T01:22:25.8338858Z ##[debug]==> azure-arm: Uploading the Windows update script...

2018-10-24T01:22:25.8341847Z ##[debug]**************************************/
2018-10-24T01:22:25.8342865Z ##[debug]Match start index: -1
2018-10-24T01:22:25.8344132Z ##[debug]Match start index: -1
2018-10-24T01:22:27.7959273Z ==> azure-arm: Running Windows update...
2018-10-24T01:22:27.7960435Z 
2018-10-24T01:22:27.7962054Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:27.7963678Z ##[debug]/*************************************
2018-10-24T01:22:27.7965399Z ##[debug]==> azure-arm: Running Windows update...

2018-10-24T01:22:27.7967272Z ##[debug]**************************************/
2018-10-24T01:22:27.7968471Z ##[debug]Match start index: -1
2018-10-24T01:22:27.7970253Z ##[debug]Match start index: -1
2018-10-24T01:22:33.0899294Z     azure-arm: Searching for Windows updates...
2018-10-24T01:22:33.0900706Z 
2018-10-24T01:22:33.0902424Z ##[debug]Parsing log line to extract output...
2018-10-24T01:22:33.0903416Z ##[debug]/*************************************
2018-10-24T01:22:33.0904384Z ##[debug]    azure-arm: Searching for Windows updates...

2018-10-24T01:22:33.0905128Z ##[debug]**************************************/
2018-10-24T01:22:33.0905805Z ##[debug]Match start index: -1
2018-10-24T01:22:33.0906429Z ##[debug]Match start index: -1
2018-10-24T01:37:10.5822659Z     azure-arm: Found Windows update (2018-10-21; 12.23 MB): Definition Update for Windows Defender Antivirus - KB915597 (Definition 1.279.208.0)
2018-10-24T01:37:10.5824221Z 
2018-10-24T01:37:10.5825926Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.5826857Z ##[debug]/*************************************
2018-10-24T01:37:10.5827779Z ##[debug]    azure-arm: Found Windows update (2018-10-21; 12.23 MB): Definition Update for Windows Defender Antivirus - KB915597 (Definition 1.279.208.0)

2018-10-24T01:37:10.5829889Z ##[debug]**************************************/
2018-10-24T01:37:10.5830744Z ##[debug]Match start index: -1
2018-10-24T01:37:10.5831428Z ##[debug]Match start index: -1
2018-10-24T01:37:10.5837678Z     azure-arm: Downloading Windows updates (1 updates; 12.23 MB)...
2018-10-24T01:37:10.5838990Z 
2018-10-24T01:37:10.5840565Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.5841455Z ##[debug]/*************************************
2018-10-24T01:37:10.5842257Z ##[debug]    azure-arm: Downloading Windows updates (1 updates; 12.23 MB)...

2018-10-24T01:37:10.5843033Z ##[debug]**************************************/
2018-10-24T01:37:10.5843705Z ##[debug]Match start index: -1
2018-10-24T01:37:10.5844344Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8730741Z     azure-arm: ERROR: Value does not fall within the expected range.
2018-10-24T01:37:10.8732401Z 
2018-10-24T01:37:10.8733895Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8734915Z ##[debug]/*************************************
2018-10-24T01:37:10.8736734Z ##[debug]    azure-arm: ERROR: Value does not fall within the expected range.

2018-10-24T01:37:10.8738421Z ##[debug]**************************************/
2018-10-24T01:37:10.8739518Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8740612Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8741732Z     azure-arm: ERROR: at <ScriptBlock>, C:\Windows\Temp\packer-windows-update.ps1: line 195
2018-10-24T01:37:10.8742290Z 
2018-10-24T01:37:10.8743083Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8743978Z ##[debug]/*************************************
2018-10-24T01:37:10.8744961Z ##[debug]    azure-arm: ERROR: at <ScriptBlock>, C:\Windows\Temp\packer-windows-update.ps1: line 195

2018-10-24T01:37:10.8746419Z ##[debug]**************************************/
2018-10-24T01:37:10.8748044Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8749005Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8756199Z     azure-arm: ERROR: at <ScriptBlock>, <No file>: line 1
2018-10-24T01:37:10.8756910Z 
2018-10-24T01:37:10.8758389Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8759598Z ##[debug]/*************************************
2018-10-24T01:37:10.8760877Z ##[debug]    azure-arm: ERROR: at <ScriptBlock>, <No file>: line 1

2018-10-24T01:37:10.8761881Z ##[debug]**************************************/
2018-10-24T01:37:10.8762880Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8763758Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8785947Z     azure-arm: ERROR EXCEPTION: System.ArgumentException: Value does not fall within the expected range.
2018-10-24T01:37:10.8786951Z 
2018-10-24T01:37:10.8788239Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8789385Z ##[debug]/*************************************
2018-10-24T01:37:10.8791143Z ##[debug]    azure-arm: ERROR EXCEPTION: System.ArgumentException: Value does not fall within the expected range.

2018-10-24T01:37:10.8792724Z ##[debug]**************************************/
2018-10-24T01:37:10.8793929Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8795232Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8796753Z     azure-arm: ERROR EXCEPTION:    at System.Management.Automation.ComInterop.ComRuntimeHelpers.CheckThrowException(Int32 hresult, ExcepInfo& excepInfo, ComMethodDesc method, Object[] args, UInt32 argErr)
2018-10-24T01:37:10.8797713Z 
2018-10-24T01:37:10.8798990Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8803206Z ##[debug]/*************************************
2018-10-24T01:37:10.8804316Z ##[debug]    azure-arm: ERROR EXCEPTION:    at System.Management.Automation.ComInterop.ComRuntimeHelpers.CheckThrowException(Int32 hresult, ExcepInfo& excepInfo, ComMethodDesc method, Object[] args, UInt32 argErr)

2018-10-24T01:37:10.8805633Z ##[debug]**************************************/
2018-10-24T01:37:10.8806312Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8806930Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8814610Z     azure-arm: ERROR EXCEPTION:    at CallSite.Target(Closure , CallSite , ComObject , Int32 )
2018-10-24T01:37:10.8815303Z 
2018-10-24T01:37:10.8816196Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8817874Z ##[debug]/*************************************
2018-10-24T01:37:10.8819235Z ##[debug]    azure-arm: ERROR EXCEPTION:    at CallSite.Target(Closure , CallSite , ComObject , Int32 )

2018-10-24T01:37:10.8824818Z ##[debug]**************************************/
2018-10-24T01:37:10.8826268Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8827155Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8828397Z     azure-arm: ERROR EXCEPTION:    at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
2018-10-24T01:37:10.8831949Z 
2018-10-24T01:37:10.8834859Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8836223Z ##[debug]/*************************************
2018-10-24T01:37:10.8837425Z ##[debug]    azure-arm: ERROR EXCEPTION:    at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

2018-10-24T01:37:10.8838494Z ##[debug]**************************************/
2018-10-24T01:37:10.8839469Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8840527Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8841389Z     azure-arm: ERROR EXCEPTION:    at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
2018-10-24T01:37:10.8841954Z 
2018-10-24T01:37:10.8842765Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8843654Z ##[debug]/*************************************
2018-10-24T01:37:10.8844715Z ##[debug]    azure-arm: ERROR EXCEPTION:    at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)

2018-10-24T01:37:10.8845793Z ##[debug]**************************************/
2018-10-24T01:37:10.8846623Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8847515Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8848516Z     azure-arm: ERROR EXCEPTION:    at System.Management.Automation.Interpreter.DynamicInstruction`3.Run(InterpretedFrame frame)
2018-10-24T01:37:10.8849166Z 
2018-10-24T01:37:10.8850576Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8851522Z ##[debug]/*************************************
2018-10-24T01:37:10.8852574Z ##[debug]    azure-arm: ERROR EXCEPTION:    at System.Management.Automation.Interpreter.DynamicInstruction`3.Run(InterpretedFrame frame)

2018-10-24T01:37:10.8853592Z ##[debug]**************************************/
2018-10-24T01:37:10.8867482Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8869283Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8906351Z     azure-arm: ERROR EXCEPTION:    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2018-10-24T01:37:10.8907207Z 
2018-10-24T01:37:10.8913644Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8916114Z ##[debug]/*************************************
2018-10-24T01:37:10.8917565Z ##[debug]    azure-arm: ERROR EXCEPTION:    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)

2018-10-24T01:37:10.8918724Z ##[debug]**************************************/
2018-10-24T01:37:10.8920154Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8921266Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8935379Z ==> azure-arm: 
2018-10-24T01:37:10.8935971Z 
2018-10-24T01:37:10.8937119Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8953881Z ##[debug]/*************************************
2018-10-24T01:37:10.8954623Z ##[debug]==> azure-arm: 

2018-10-24T01:37:10.8955314Z ##[debug]**************************************/
2018-10-24T01:37:10.8956311Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8956956Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8957441Z ==> azure-arm: Cleanup requested, deleting resource group ...
2018-10-24T01:37:10.8957907Z 
2018-10-24T01:37:10.8958550Z ##[debug]Parsing log line to extract output...
2018-10-24T01:37:10.8959414Z ##[debug]/*************************************
2018-10-24T01:37:10.8960185Z ##[debug]==> azure-arm: Cleanup requested, deleting resource group ...

2018-10-24T01:37:10.8960907Z ##[debug]**************************************/
2018-10-24T01:37:10.8961572Z ##[debug]Match start index: -1
2018-10-24T01:37:10.8962212Z ##[debug]Match start index: -1
2018-10-24T01:41:46.5541340Z ==> azure-arm: Resource group has been deleted.
2018-10-24T01:41:46.5542779Z ```

**Comment by tvories** [comment]
This should be the same as #24 which affects Server 2008.

**Comment by tvories** [comment]
This appears to be fixed with [0.61](https://github.com/rgl/packer-provisioner-windows-update/releases/tag/v0.6.1)!

**Comment by rgl** [comment]
will close this since it seems to be fixed in 0.6.1 :-)
